121 - Get demo-able project working

cowmanjoe opened the pull request:
This is a PR to try to get our project front-end to work properly with the back end. I have set the base of the PR as feature/110 because that is the latest work I've done on the server side. We should make the base develop after we merge in feature/110 when it is approved. I hackily got it to work on my machine, i.e. clicking search gives search results that are from the database. 

It would be great if some people who are more familiar with the front end could take a look and fix some of the problems. Several things should probably be fixed: 
1. The home page and the results are on the same page, I don't know how the Router object works and I just hard coded in the ResultsTable into the App component
2. Redux is no longer used, I commented out some parts that made Redux control state 
3. The ResultData object does not match what is in our database, look on the wiki for what the database schema looks like

Number 1 is the main issue that is actually visible, so it is high priority. The will be unaffected if 2 and 3 are not fixed. 

Also PLEASE TEST at the very least if possible!

kimoantiqe committed:
Setup docker to spin up db

kimoantiqe committed:
Update README.md

kimoantiqe committed:
added more details about containers

kimoantiqe committed:
Added details on how to connect to db

kimoantiqe committed:
added useful docker-compose commands

kimoantiqe committed:
Update README.md

kimoantiqe committed:
added init script to run cowan's db dump data

kimoantiqe committed:
Merge branch 'feature/113' of https://github.com/ubclaunchpad/Internado into feature/113

cowmanjoe committed:
Got client to server search functionality working, pretty hacky though.

frostyshadows commented:
Looks like Number 1 might be related to this? https://github.com/ubclaunchpad/Internado/issues/96

cowmanjoe commented:
Yea, I think you’re right. There appeared to be some solution using the Router, but I hadn’t seen it work and I essentially worked around it for the initial commit in this PR.

cormacjmollitor committed:
Able to Search For Jobs for Demo

The navbar is now properly hooked up to the backend to display returned postings from search queries. It doesn't use Redux so it's hacky for now but it's sufficient for our demo

cormacjmollitor commented:
I pushed a change for solving number 1. I'll make a change for number 3 tomorrow morning and update the ResultRow component and the ResultData class

MichaelXian commented:
Made the ResultsData match the database schema, and the table displays it properly from what I can tell.

cowmanjoe commented:
@cormacjmollitor For me, the search button doesn't redirect to /search which I think it's supposed to do if I'm not mistaken. Does it do that for you?

cormacjmollitor commented:
@cowmanjoe It only works from "localhost:3000/results". I'm looking into routing to "localhost:3000/results" from the landing page on search now

cormacjmollitor committed:
Now redirects to results table on search

cormacjmollitor commented:
@cowmanjoe I just pushed a change for redirecting to "localhost:3000/results"

cowmanjoe commented:
Awesome! Everything seems to be working well on my laptop!

cowmanjoe commented:
One thing if you have time: the home button doesn't route back to the home page. Not a huge issue but I feel like this should be a quick fix.

kimoantiqe committed:
Updated to latest pg_dump

kimoantiqe committed:
added more details about hosts

kimoantiqe committed:
added details on how to init db with data

cowmanjoe commented:
Should we merge this into develop? I commented out our use of redux but I think @cormacjmollitor said that we may not even need to be using it at the moment and I tend to agree. We tested it and used it in the demo for connecting the database, server, and client together. 

@frostyshadows @AmirNaghibi

kimoantiqe committed:
Merge branch 'develop' into feature/113

kimoantiqe committed:
Updated dump file

kimoantiqe committed:
Merge branch 'feature/113' of https://github.com/ubclaunchpad/Internado into feature/113

frostyshadows commented on client/src/components/ResultRow.js:
```diff
@@ -13,10 +13,10 @@ class ResultRow extends Component {
         break;
       case 'post': // Clicked on original post
-        window.open(result.postUrl, '_blank');
+        window.open(result.link, '_blank');
```
should add a check to make sure link isn't empty

cowmanjoe replied:
Good idea. I just made it so if the link is null, undefined or the empty string then the linke doesn't do anything. In the future maybe we could have some mechanism of informing us of the broken link (or even attempting to retrieve the missing link).

frostyshadows commented on client/src/App.js:
```diff
@@ -1,22 +1,23 @@
-            ],
-            isFilterMenuVisible: false
+            data: [],
+            // data: [
```
I would recommend deleting this from this file instead of commenting it out, and putting it in a separate file since we may use it for testing in the future.

cowmanjoe replied:
I just deleted the commented code entirely and didn't put it into another file. I couldn't find a place I thought made sense for it, and I figured that the data isn't that hard to generate since it was all made up and there wasn't that much of it. Plus, stuff may change before we write tests, and we can also easily go back and copy the data in previous commits if needed.

frostyshadows reviewed (changes requested):
Yeah Redux shouldn't be necessary for our MVP since we're not keeping that much state, and React allows for some state management already. My opinion is keep it commented out for now since it may be useful later on.

cowmanjoe committed:
Added check to make sure link isn't empty and deleted old dummy data.

cowmanjoe committed:
Merge branch 'develop' into feature/121

cowmanjoe committed:
Added error message for failing to create PGSQL connection.

cowmanjoe committed:
Merge branch 'feature/113' into feature/121

cowmanjoe committed:
Used environment variables for TypeORM config when using Docker.

This was because the location for the server is assigned by Docker and it is no longer localhost, so we must use this value and assign it to the TYPEORM_HOST environment variable for the server image.

cowmanjoe commented:
I merged in Karim's #113 branch (feature/113) to combine into a single PR because I wanted to make sure that it all worked together. We can revert the commits back if the PR is now too large/broad. 

I've tested this on my machine and it all appears to work.

cowmanjoe committed:
Corrected copy command to use tabs as separators for the COPY command.

frostyshadows commented on client/src/App.js:
```diff
@@ -35,21 +31,43 @@ class App extends Component {
+      }
+
+      console.log(body);
+			this.setState({ data: body.result });
```
spacing

frostyshadows commented on client/src/App.js:
```diff
@@ -1,22 +1,18 @@
-            ],
-            isFilterMenuVisible: false
+            data: [],
+						isFilterMenuVisible: false
```
spacing

frostyshadows reviewed (approved):
LGTM! Approved, but could you fix the 2 spacing issues before merging? I had that happen on Github too, it had something to do with VSCode and Github displaying tab/spaces differently from each other.

cowmanjoe commented:
@frostyshadows Sorry I didn't see your comments before merging! I made a small PR for this: #140.
